CMakeLists.txt

```diff
@@ -249,6 +249,50 @@ else()
     message(FATAL_ERROR "${CMAKE_CXX_COMPILER_ID} not yet supported.")
 endif()
 
+# To improve output determinism enable precise floating point operations globally
+# This code was based on lib/astc-encoder/Source/cmake_core.cmake
+
+# For Visual Studio prior to 2022 (compiler < 19.30) /fp:strict
+# For Visual Studio 2022 (compiler >= 19.30) /fp:precise
+# For Visual Studio 2022 ClangCL seems to have accidentally enabled contraction by default,
+# so behaves differently to CL.exe. Use the -Xclang argument to workaround and allow access
+# GNU-style switch to control contraction and force disable.
+
```

Comment:
Add this too

```
# On CMake 3.25 or older CXX_COMPILER_FRONTEND_VARIANT is not always set
if(CMAKE_CXX_COMPILER_FRONTEND_VARIANT STREQUAL "")
    set(CMAKE_CXX_COMPILER_FRONTEND_VARIANT "${CMAKE_CXX_COMPILER_ID}")
endif()
```

3.25 is not that old so people may be using it.

Reply:
Fair point, added it as requested